README.md

```diff
@@ -66,139 +69,222 @@ or
-* C / C++ / C# / CSS
-* Elixir
-* Go
-* HTML
-* Java / JavaScript / JSX / JSON / Julia
-* Nix
-* PHP / Python
-* R / Ruby / Rust
-* Scala / Swift
-* TypeScript / TSX
+- Bash
+- C / C++ / C# / CSS
+- Elixir
+- Go
+- HTML
+- Java / JavaScript / JSX / JSON / Julia
+- Nix
+- PHP / Python
+- R / Ruby / Rust
+- Scala / Swift
+- TypeScript / TSX
 
 These languages are in development:
 
-* Agda
-* Elm
-* Emacs Lisp
-* OCaml
-* XML (upstream)
+- Agda
+- Elm
+- Emacs Lisp
+- OCaml
+- XML (upstream)
 
-## ⚖️ Indicators Mode
+## 📝 Customization
 
-<p align="center">
-<img src="./etc/indicators.png" width="40%" height=480%"/>
-</p>
+Although ts-fold aims to have good folding out of the box for all supported definitions, people will indubitably have their own preferences or desired functionality. The following section outlines how to add your own folding definitions and folding functions to make ts-fold work for you. If there are any improvements you find for existing or new languages, please do raise a PR so that others may benefit from better folding in the future!
 
-You need to load `ts-fold-indicators-mode`:
-- `use-package`
-   ```elisp
-   (use-package ts-fold-indicators
-   :straight (ts-fold-indicators :type git :host github :repo "emacs-tree-sitter/ts-fold"))
-   ```
+### :white_circle: Folding on new nodes
 
--
-   ```elisp
-   (add-to-list 'load-path "/path/to/lib")
-   (require ts-fold)
-   ```
-   or
+Ts-fold defines all its folding definitions in the the variable `ts-fold-range-alist` which is an alist with the key of the alist being the mode and the value being another alist of fold definitions.
 
-   ```elisp
-   (use-package ts-fold-indicators
-      :load-path "/path/to/lib")
-   ```
+```elisp
+;; Example of ts-fold-range-alist's structure
+'((c-mode . c-folding-definitions) ;; *-folding-definitions is structured as shown below
+  (css-mode . css-folding-definitions)
+  (go-mode . go-folding-definitions)
+  (scala-mode . scala-folding-definitions)
+  ...)
+
+;; Examle of a folding definition alist
+(setq css-folding-definitions
+    (block . ts-fold-range-seq)
+    (comment . ts-fold-range-c-like-comment))
+```
 
-You can then enable this manually by doing the following
+So you can select whatever node that you want to fold on it.
 
-```
-M-x ts-fold-indicators-mode
-```
+To find what node you'll want to fold closed, refer to the (tree-sitter documentation)[https://emacs-tree-sitter.github.io/getting-started/#view-the-syntax-tree] about viewing nodes. `tree-sitter-debug` and `tree-sitter-query-builder` are both very useful for this.
 
-- To enable this automatically whenever `tree-sitter-mode` is enabled:
+For the folding functions, ts-fold provides some default
 
-   ```el
-   (add-hook 'tree-sitter-after-on-hook #ts-fold-indicators-mode)
-   ```
+- `ts-fold-range-seq` - Folds from the start of the node to the end of the node leaving a buffer of one character on each side. Usually used for code blocks that have bracketing delimiters.
```

Comment:
I prefer max 80 columns per line, can we avoid these long lines?

Sorry, I should have added the `.dir-locals.el`. 😅

Reply:
No problem, I usually right in org mode so I just let the lines go as long as they want 😅.